File: app/src/jvx/java/art/coded/wireframe/viewmodel/PagingViewModel.java
package art.coded.wireframe.viewmodel;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import androidx.paging.PagedList;

import art.coded.wireframe.model.entity.Element;
import art.coded.wireframe.model.ElementRepository;

public class PagingViewModel extends ViewModel {

    private final String LOG_TAG = PagingViewModel.class.getSimpleName();

    ElementRepository mRepository;

    public void loadData(Context context) {
        mRepository = new ElementRepository(context);
    }

    public void removeData(Element element) { mRepository.delete(element); }
    public LiveData<PagedList<Element>> elementList() {
        return (mRepository.getPagedList());
    }
}

File: app/src/ktx/java/art/coded/wireframe/viewmodel/PagingViewModel.kt
package art.coded.wireframe.viewmodel

import androidx.lifecycle.*
import androidx.paging.PagedList
import art.coded.wireframe.model.ElementRepository
import art.coded.wireframe.model.entity.Element
import kotlinx.coroutines.launch

private val LOG_TAG = PagingViewModel::class.java.simpleName

class PagingViewModel(private val repository: ElementRepository) : ViewModel() {

    val actionPending = MutableLiveData<Boolean>()

    fun getData(): LiveData<PagedList<Element>> {
        var elements: LiveData<PagedList<Element>>? = null
        actionPending.postValue(true)
        viewModelScope.launch {
            elements = repository.pagedList
        }
        actionPending.postValue(false)
        return elements!!
    }

    fun removeData(element: Element) {
        viewModelScope.launch {
            repository.delete(element)
        }
    }

    class PagingViewModelFactory(private val repository: ElementRepository) : ViewModelProvider.Factory {
        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            return PagingViewModel(repository) as T
        }
    }
}